enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1448,14 +1449,33 @@ func (p *PebbleCache) userGroupID(ctx context.Context) string {
 	return user.GetGroupID()
 }
 
-func (p *PebbleCache) lookupGroupAndPartitionID(ctx context.Context, remoteInstanceName string) (string, string) {
+func (p *PebbleCache) lookupGroupAndPartitionID(ctx context.Context, remoteInstanceName string) (string, string, error) {
 	groupID := p.userGroupID(ctx)
+	requestedPartition := cacheutil.PartitionOverride(ctx)
+
+	if requestedPartition == DefaultPartitionID {
+		return groupID, DefaultPartitionID, nil
+	}
+
 	for _, pm := range p.partitionMappings {
-		if pm.GroupID == groupID && strings.HasPrefix(remoteInstanceName, pm.Prefix) {
-			return groupID, pm.PartitionID
+		// If a partition override was requested, check for a match.
+		matchesPartitionOverride := requestedPartition != "" &&
```

Comment:
does this mean that remote instance name is ignored for partition overrides? Is this what we want (just checking)

Reply:
Are you referring to this portion on L1473? `strings.HasPrefix(remoteInstanceName, pm.Prefix)`

I'm not positive what the intended use case for that code was, but we don't set any prefixes currently 
https://github.com/buildbuddy-io/buildbuddy-internal/blob/master/enterprise/config/us-west1/buildbuddy.prod.yaml#L177